src/asf_tools/watermasking/generate_worldcover_tiles.py

```diff
@@ -0,0 +1,273 @@
+    """Get a list of the Worldcover tile filenames that are necessary to overlap an OSM tile.
+
+    Args:
+        osm_tile: The lower left corner (lat, lon) of the desired OSM tile.
+        wc_tile_width: The width of the Worldcover tiles in degrees.
+        tile_width: The width of the OSM tiles in degrees.
+
+    Returns:
+        filenames: The list of Worldcover filenames.
+    """
+    filenames = []
+    tiles = get_tiles(osm_tile_coord, wc_tile_width, tile_width)
+    for tile in tiles:
+        filenames.append(worldcover_tile_dir + lat_lon_to_tile_string(tile[0], tile[1], is_worldcover=True))
+    return filenames
+
+
+def crop_tile(tile, lat, lon, tile_width, tile_height):
+    """Crop the merged tiles
+
+    Args:
+        tile: The filename of the desired tile to crop.
+    """
+    in_filename = UNCROPPED_TILE_DIR + tile
+    out_filename = CROPPED_TILE_DIR + tile
+    pixel_size_x, pixel_size_y = 0.00009009009, -0.00009009009
+
+    src_ds = gdal.Open(in_filename)
+    gdal.Translate(
+        out_filename,
+        src_ds,
+        projWin=[lon, lat+tile_height, lon+tile_width, lat],
+        xRes=pixel_size_x,
+        yRes=pixel_size_y,
+        outputSRS='EPSG:4326',
+        format='COG',
+        creationOptions=['NUM_THREADS=all_cpus']
+    )
+    remove_temp_files(['tmp_px_size.tif', 'tmp.shp'])
+
+
+def build_dataset(worldcover_tile_dir, lat_range, lon_range, tile_width, tile_height):
+    """ Main function for generating a dataset with worldcover tiles.
+
+    Args:
+        worldcover_tile_dir: The directory containing the unprocessed worldcover tiles.
+        lat_range: The range of latitudes the dataset should cover.
+        lon_range: The range of longitudes the dataset should cover.
+        out_degrees: The width of the outputed dataset tiles in degrees.
+    """
+    for lat in lat_range:
+        for lon in lon_range:
+            start_time = time.time()
+            tile = lat_lon_to_tile_string(lat, lon, is_worldcover=False)
+            tile_filename = UNCROPPED_TILE_DIR + tile
+            worldcover_tiles = lat_lon_to_filenames(worldcover_tile_dir, (lat, lon), WORLDCOVER_TILE_SIZE, tile_width)
+            print(f'Processing: {tile_filename} {worldcover_tiles}')
+            merge_tiles(worldcover_tiles, tile_filename, 'GTiff', compress=True)
+            crop_tile(tile, lat, lon, tile_width, tile_height)
+            end_time = time.time()
+            total_time = end_time - start_time
+            print(f'Time Elapsed: {total_time}s')
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        prog='generate_worldcover_tiles.py',
+        description='Main script for creating a tiled watermask dataset from the ESA WorldCover dataset.'
+    )
+
+    parser.add_argument('--worldcover-tiles-dir', help='The path to the directory containing the worldcover tifs.')
+    parser.add_argument('--lat-begin', help='The minimum latitude of the dataset.', default=-85, required=True)
```

Comment:
I think overall things look good, but I had some questions on this: 
1. Is minimum and maximum long/lat how other places refer to extensions? Does minimum mean the lowest number? Does this have to be in one projection? 
2. Is there a reason you name this parameter lat/long_begin and lat/long_end instead of min/max? 

Small picky details, sorry!